src/Concordium/Client/Types/Contract/Info.hs

```diff
@@ -281,10 +313,32 @@ constructModuleInspectInfo namedModRef wasmVersion moduleSchema exportedFuncName
 
             mis1ContractSigs = insertReceiveNames funcNames cSigsWithoutReceives
         in ModuleInspectSigsV1{..}
+      SchemaV2 ->
+        let mkContrSchemaTuples x xs = case x of
+              CS.InitFuncName contrName -> (contrName, ContractSigsV2 { csv2InitSig = Nothing
+                                                                      , csv2ReceiveSigs = Map.empty
+                                                                      }) : xs
+              CS.ReceiveFuncName _ _ -> xs
+            cSigsWithoutReceives = Map.fromList . foldr mkContrSchemaTuples [] $ funcNames
+
+            insertReceiveNames :: [CS.FuncName] -> Map.Map Text ContractSigsV2 -> Map.Map Text ContractSigsV2
+            insertReceiveNames [] sigMap = sigMap
+            insertReceiveNames (CS.InitFuncName _:remaining) sigMap = insertReceiveNames remaining sigMap
+            insertReceiveNames (CS.ReceiveFuncName cname fname:remaining) sigMap = case Map.lookup cname sigMap of
+              Nothing -> insertReceiveNames remaining sigMap -- This should never happen, as we validate modules before they are put on chain.
```

Comment:
I am not sure this is correct. I don't think we do any validation on the chain for schemas, so in principle this can happen.